tests/Unit/Parallel/CMakeLists.txt

```diff
@@ -155,6 +156,31 @@ function(add_algorithm_test_with_balancing TEST_NAME EXECUTABLE_NAME REGEX_TO_MA
   endif()
 endfunction()
 
+# Run TEST_NAME twice: first with no command-line args, then with the
+# `+restart SpectreCheckpoint0` command-line arg to restart from a Charm++
+# checkpoint named `SpectreCheckpoint0`. The checkpoint file is then deleted
+# to avoid it getting in the way of future test invocations (which would try
+# to overwrite the prior checkpoint file and error).
+function(add_algorithm_test_with_restart_from_checkpoint TEST_NAME)
+  # Executable writes checkpoint file into the dir where it runs:
+  set(CHECKPOINT "${CMAKE_BINARY_DIR}/tests/Unit/Parallel/SpectreCheckpoint0")
+  add_test(
+    NAME "\"Unit.Parallel.${TEST_NAME}\""
+    COMMAND
+    ${SHELL_EXECUTABLE}
+    -c
+    "${CMAKE_BINARY_DIR}/bin/Test_${TEST_NAME} \
+    && ${CMAKE_BINARY_DIR}/bin/Test_${TEST_NAME} +restart ${CHECKPOINT} \
+    && rm -r ${CHECKPOINT} 2>&1"
```

Comment:
Also do the deletion at the start, so running the test after a previous failure works.